### PR TITLE
LSIF: Rename fields in ApiJob and LSIFJob payloads.

### PR DESCRIPTION
**Pull request by efritz**
Rename fields of LSIF job payloads in GraphQL and LSIF-server API to be more consistent with existing types.

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -253,8 +253,10 @@ README.md @sqs
 /cmd/frontend/db/discussion* @slimsag
 
 # LSIF
-/cmd/frontend/internal/httpapi/lsif.go @sourcegraph/code-intel
 /lsif/ @sourcegraph/code-intel
+/cmd/frontend/graphqlbackend/*lsif* @sourcegraph/code-intel
```
Comment (unknwon) [inline .github/CODEOWNERS]: Do you think it is more precise?

```suggestion
/cmd/frontend/graphqlbackend/*lsif_* @sourcegraph/code-intel
```

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 2]
```diff
@@ -3980,37 +3980,40 @@ type LSIFJobStats implements Node {
...
-# A queued, active, or completed LSIF job.
+# Metadata and status about an LSIF job.
 type LSIFJob implements Node {
     # The ID.
     id: ID!
 
-    # The job type (convert, or clean-old-jobs).
-    name: String!
+    # The job type.
+    jobType: String!
```
Comment (felixfbecker) [inline cmd/frontend/graphqlbackend/schema.graphql]: This may be redundant since it's a field on `Job`

Comment (efritz) [reply cmd/frontend/graphqlbackend/schema.graphql]: Would there be a conflict naming it `type` as a reserved word in graphql/go/ts? What would you prefer for this kind of name?

Comment (felixfbecker) [reply cmd/frontend/graphqlbackend/schema.graphql]: I don't know about Go, but `type` is a valid identifier in TS and GraphQL, so I'd use that

Comment (efritz) [reply cmd/frontend/graphqlbackend/schema.graphql]: Updated, see no problems.

**Pull request merged**